plugins/obs-qsv11/QSV_Encoder.cpp

```diff
@@ -81,18 +81,18 @@ qsv_t *qsv_encoder_open(qsv_param_t *pParams, enum qsv_codec codec)
 {
 	obs_video_info ovi;
```

Comment:
This whole block down to initializing the internal and opening the encoder is windows specific, and `adapters` is queried in the platform specific `check_adapters` function.

Your change is mostly trying to plumb this data down into the platform specific `Initialize` implementation.

It seems it would be easier to move this code into common_utils_windows.cpp's Initialize and storing what you need in static globals (for example a pointer `adapters` array would be fine, we know its only initialized once at module load). Then you dont need to plumb anything, and we can make some later refactors to qsv more easily.

Reply:
This should help resolve the issue with not having an adapter idx during `::Reset` as well.

Reply:
Thank you so much for this suggestion! It should be fixed now.